janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -468,18 +469,31 @@ private static Object convertToEsType(Object value) {
         } else throw new IllegalArgumentException("Unsupported type: " + value.getClass() + " (value: " + value + ")");
     }
 
-    private static Object convertgeo(Geoshape geoshape) {
-        if (geoshape.getType() == Geoshape.Type.POINT) {
+    @SuppressWarnings("unchecked")
+    private static Object convertgeo(Geoshape geoshape, Mapping mapping) {
+        if (geoshape.getType() == Geoshape.Type.POINT && Mapping.PREFIX_TREE != mapping) {
             Geoshape.Point p = geoshape.getPoint();
             return new double[]{p.getLongitude(), p.getLatitude()};
-        } else if (geoshape.getType() != Geoshape.Type.BOX && geoshape.getType() != Geoshape.Type.CIRCLE) {
+        } else if (geoshape.getType() == Geoshape.Type.BOX) {
```

Comment:
Yes it's OK.
IOException is throw by _geoshape.toMap()_

For Box, the map is constructed without using _geoshape.toMap()_.
For Circle, the map is constructed with _geoshape.toMap()_.